packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/shouldAttachGeoData.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  TransactionInstruction,
+} from '@solana/web3.js'
+import { config } from '../../config'
+
+const MEMO_PROGRAM_ID = 'Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo'
+const MEMO_V2_PROGRAM_ID = 'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr'
+const PAYMENT_ROUTER_PROGRAM_ID = config.paymentRouterProgramId
+
+const RECOVERY_MEMO_STRING = "Recover Withdrawal"
+
+/**
+ * Determines whether or not the provided instructions are from a payment transaction
+ * @param instructions the instructions to check
+ * @returns boolean
+ */
+const isPaymentTransaction = (
+  instructions: TransactionInstruction[]
+) => {
+  let usesPaymentRouter = false
+  let isRecovery = false
+  for (let i = 0; i < instructions.length; i++) {
+    const instruction = instructions[i]
+    switch (instruction.programId.toBase58()) {
+      case PAYMENT_ROUTER_PROGRAM_ID:
+        usesPaymentRouter = true
+        break
+      case MEMO_PROGRAM_ID:
+      case MEMO_V2_PROGRAM_ID: {
+        console.log(instruction.data)
+        const memo = instruction.data.toString()
+        if (memo === RECOVERY_MEMO_STRING) {
+          isRecovery = true
+        }
+        break
+      }
+      default:
+        break
+    }
+  }
+  return usesPaymentRouter && !isRecovery
```

Comment:
It would probably be safer to check for memos that match valid purchase formats (starts with `track:` or `album:` if I'm not mistaken). This puts us at the risk of thinking any future transactions which happen to use payment router are purchases.

Reply:
I started doing this change (quite easy) but then realized we also include `playlist:` on the indexing side and it seems more likely than not we'll forget to add types here and I don't think relay should even need to care. I don't see any harm with our relay attaching this geo data even if it doesn't get indexed

Reply:
I missed that this was not the indexer 🤦 . Carry on.